--- /var/tmp/portage/x11-apps/xinit-1.3.4-r1/work/xinit-1.3.4/startx.cpp	2014-09-11 21:31:42.000000000 +0400
+++ /tmp/startx.cpp	2017-05-10 09:18:27.257568907 +0300
@@ -269,7 +269,7 @@
     dummy=0
 
     XCOMM create a file with auth information for the server. ':0' is a dummy.
-    xserverauthfile=$HOME/.serverauth.$$
+    xserverauthfile=$XAUTHORITY
     trap "rm -f '$xserverauthfile'" HUP INT QUIT ILL TRAP KILL BUS TERM
     xauth -q -f "$xserverauthfile" << EOF
 add :$dummy . $mcookie
